### ISSUE TITLE
Writing long UTF encoded Strings into PactRecords

### ISSUE THREAD

**Issue by mleich** [stratosphere/stratosphere#294]
I've implemented my own data type that represents a web document (URL, text content, some metadata).
When serializing this document into PactRecord I use writeUTF to write the text content.

I often encounter documents that are longer than 65535 bytes in UTF encoding.

https://github.com/stratosphere/stratosphere/blob/master/pact/pact-common/src/main/java/eu/stratosphere/pact/common/type/PactRecord.java#L1564

What can I do to fix that?

**Comment by StephanEwen** [comment]
The writeUTF method is specified in the jaava DataOutput interface to
encode size in 2 bytes. PactString uses a different way of representing its
data, for that exact reason.

I do not think we can change writeUTF without breaking some functionality
(especially the aspired Hadoop compatibility). Can you use the PactString
Logic? Maybe refactor the PactString such that you have a static utility
method that lets you write char sequence or so to the stream?

**Comment by StephanEwen** [comment]
Since we cannot change the writeUTF method (its functionality is specified by the java interface), I have added static utility serialization methods to PactString, that support long string serialization.

```
public class MyLongStringType {
    private String theString;

    public void read(DataInput in) {
        theString = PactString.readString(in);
    }

    public void write(DataOutput out) {
        PactString.writeString(theString, out);
    }
}

Fixed in 38725c715924a38432b3c16d0e781d630b7b0500
```

**Comment by StephanEwen** [comment]
If this solution does not cover your case, please re-open this issue!
